### ISSUE TITLE
Question about model parameter size

### ISSUE THREAD

**Issue by pablogranolabar** [guxd/DialogBERT#6]
I am interested in implementing gradient checkpointing to support DialogBERT-XL training. What would the level of effort be with modifying DialogBERT to support an equivalent parameter size as GPT2-XL?

Thanks in advance!

**Comment by guxd** [comment]
I think it won't take much effort if you just modify the hyperparameter setting in model initialization.
However, you need much effort to finetune your model from scratch. 

One more issue is that you need a large-scale dialogue dataset for finetuning.

**Comment by pablogranolabar** [comment]
When you say much effort to fine tune, do you mean that the current process when invoking main.py does not complete the full fine tune run? For example, with:
```
python3 main.py --model_size=large --per_gpu_train_batch_size=128
```
..is that fully training the Large parameter model or is additional conversational set fine tune training needed to obtain the results in your research?  

If the changes are only to the parameter size to get to GPT2-XL, would the current process work for fine tuning?

**Comment by guxd** [comment]
The current main.py has already completed the code, but you need much effort to finetune hyperparameters in your dataset.

**Comment by pablogranolabar** [comment]
I am curious, why wouldn't the current hyperparameters used for the large parameter model work with a larger parameter size? The current large model has already been through hyperparameter tuning I assume?

**Comment by guxd** [comment]
Once you have changed your dataset, you need to finetune your hyperparameter again.

**Comment by pablogranolabar** [comment]
So does the current training process using main.py train on the complete dataset that you obtained the paper results with? Or do we need to add a conversational dataset to the training pipeline such as the DialoGPT Reddit conversational dataset?

**Comment by guxd** [comment]
You do not need any extra dataset to obtain the results. 
Finetuning from dailydial or multiwoz won't alleviate the overfitting issue I think.
Training in a larger dataset might alleviate.
